scripts/exglobal_diag.sh

```diff
@@ -170,113 +170,126 @@ EOFdiag
    #        innovation files.
 
    loops="01 03"
-   for loop in $loops; do
-      case $loop in
+   for loop in ${loops}; do
+      case "${loop}" in
          01) string=ges;;
          03) string=anl;;
-          *) string=$loop;;
+          *) string="${loop}";;
       esac
-      echo $(date) START loop $string >&2
+      echo "$(date)" START loop "${string}" >&2 || true
       n=-1
-      while [ $((n+=1)) -le $ntype ] ;do
-         for type in $(echo ${diagtype[n]}); do
-            count=$(ls ${prefix}${type}_${loop}* 2>/dev/null | wc -l)
-            if [ $count -gt 1 ]; then
-               if [ $USE_CFP = "YES" ]; then
-                  echo "$nm $DATA/diag.sh $lrun_subdirs $binary_diag $type $loop $string $CDATE $DIAG_COMPRESS $DIAG_SUFFIX" | tee -a $DATA/mp_diag.sh
-                  if [ ${CFP_MP:-"NO"} = "YES" ]; then
+      # shellcheck disable=SC2116
+      # shellcheck disable=SC2012
+      # shellcheck disable=SC2003
```

Comment:
@WalterKolczynski-NOAA If I disable these I have no means of testing whether the application runs as it is supposed to do. If anyone needs to correct these, it should be Cory Martin since he is the original author. That, or Russ Treadon could test since he identified the problem.

Reply:
Disabling shellcheck warnings has zero impact on how a script runs. Shellcheck is not involved in the running of a script.

Reply:
I understand that and I apologize that I misspoke. 

What I meant was that without those shellchecks, the `shellnorms` action will not pass.

Reply:
Let's see that first. It shouldn't flag stuff unless you added errors, but I also went through and made suggestions for all the complaints I recognized by eye (check behind the fold, there are still a bunch you haven't updated yet).

As for testing, you can compare against a run using develop and make sure the output is bitwise identical.